File: TwoPointers/container-with-most-water/container-with-most-water.cs
public class Solution
{
    public int MaxArea(int[] height)
    {
        int l = 0;
        int r = height.Length - 1;
        int res = 0;

        while (l < r)
        {
            int area = Math.Min(height[l], height[r]) * (r - l);
            res = Math.Max(area, res);

            if (height[l] <= height[r])
            {
                l++;
            }
            else
            {
                r--;
            }
        }

        return res;
    }
}

File: TwoPointers/container-with-most-water/container-with-most-water.js
/**
 * @param {number[]} height
 * @return {number}
 */
var maxArea = function (height) {
    let l = 0;
    let r = height.length - 1;
    let res = 0;

    while (l < r) {
        const area = Math.min(height[l], height[r]) * (r - l);
        res = Math.max(res, area);

        if (height[l] <= height[r]) {
            l++;
        } else {
            r--;
        }
    }

    return res;
};
